src/script/lua_api/l_client.cpp

```diff
@@ -169,6 +174,11 @@ int ModApiClient::l_get_node(lua_State *L)
 {
 	// pos
 	v3s16 pos = read_v3s16(L, 1);
+
+	if (getClient(L)->getCSMFlavourLimits() & CSMFlavourLimit::CSM_FL_LOOKUP_NODES) {
+		// TODO
```

Comment:
it seems you don't understand what is performance, especially when code doesn't exist how can judge it ? 
Imagine you have too many imagination and imagine wrong things @paramat , oh, it's what you do.